### DISCUSSION TITLE
Enable "Select a map to load" screen

### DISCUSSION THREAD

**Discussion by tomakela** [sfall-team/sfall#501] [Ideas]
Is there a way to launch (or do you plan to implement in sfall) FO2 "Select map to load" -dialog without editing the executable?

<img width="110" alt="image" src="https://github.com/sfall-team/sfall/assets/9822663/8c7c2f8c-6a10-4c1a-873c-f6fae42259ac">

Thanks for your excellent work!

**Comment by NovaRain** [comment]
Pressing Ctrl+R on the main menu should bring up the dialog.
What do you need this for? If you want to check game maps, better use the mapper (mapper2.exe) instead.

**Comment by tomakela** [comment]
This is a different piece of code (the title says "Select a map to load" and with the ctrl+R/recording screen says "Select map") that allows you to in-game "teleport" into any map with the current character and continue playing normally. Essentially it skips the world map travel. Seems to be left over debug code, not accessible in the final build. Found it some 20 years ago :) https://www.nma-fallout.com/threads/fallout-fallout2-jump-to-any-map-patch.154862/ . And it probably has some side effects.

Possible use cases:
-maybe circumventing if the save game has a glitched exiting/entering a map?
-help with glitched world map travel? (or perhaps somehow help to retrieve a vanished highwayman)
-being stuck behind an NPC (This was mostly FO1 thing I think)
-fun-removing-and-therefore-very-evil cheating :(
-it could help with modding(/debugging/tinkering/fooling around) as you can change between maps without needing to build travel logic between them

**Comment by Lexx2k** [comment]
Sounds like it would be easier to just write a cheat script.

**Comment by tomakela** [comment]
And you are absolutely correct. I didn't realize that it was that easy. For those interested

in ddraw.ini set
```
AllowUnsafeScripting=1
```
And compile something along the lines (worked for Fallout 2 1.02 steam version):
```
/*
Modified from gl_uihotkeys.ssl
*/

#include "..\headers\sfall\sfall.h"
#include "..\headers\sfall\dik.h"

procedure start;

procedure start begin
   if game_loaded then begin
      register_hook(HOOK_KEYPRESS);
   end else begin
      variable
         event := get_sfall_arg,
         keyDX := get_sfall_arg,
         dummy;

      if (event) then begin
         if (keyDX == DIK_J) then begin
            dummy = call_offset_r0(0x4829F0);
         end
      end
   end
end
 
```

Thanks for the tip!
